### PR TITLE
node/governor: address ToB findings

### PR DESCRIPTION
**Pull request by tbjump**
* document usage of mutex (TOB-WORMGUWA-2)
* fix potential nil ptr deref in reloadPendingTransfer (TOB-WORMGUWA-3)
* add type assertion in queryCoinGecko (TOB-WORMGUWA-4)

### CODE REVIEW

**Comment by SEJeff** [comment]
Consider closing this PR as it is a subset of #2850

**Comment by tbjump** [comment]
Sorry, these commits should not have been included in https://github.com/wormhole-foundation/wormhole/pull/2850. I removed them from https://github.com/wormhole-foundation/wormhole/pull/2850, so this PR should still remain.

File: node/pkg/governor/governor_db.go [hunk 1]
```diff
@@ -89,6 +89,10 @@ func (gov *ChainGovernor) reloadPendingTransfer(pending *db.PendingTransfer, now
 
 	payload, err := vaa.DecodeTransferPayloadHdr(msg.Payload)
 	if err != nil {
+		if payload == nil {
```
Comment (bruce-riley) [inline node/pkg/governor/governor_db.go:92]: I think the real fix for this is to just delete lines 101 and 102 (don't reference payload on error).

**Review by bruce-riley — APPROVED** [review]

File: node/pkg/governor/governor.go [hunk 2]
```diff
@@ -117,14 +117,14 @@ func (ce *chainEntry) isBigTransfer(value uint64) bool {
...
 	mutex                 sync.Mutex
-	tokens                map[tokenKey]*tokenEntry
-	tokensByCoinGeckoId   map[string][]*tokenEntry
-	chains                map[vaa.ChainID]*chainEntry
-	msgsSeen              map[string]bool // Key is hash, payload is consts transferComplete and transferEnqueued.
-	msgsToPublish         []*common.MessagePublication
+	tokens                map[tokenKey]*tokenEntry     // protected by `mutex`
+	tokensByCoinGeckoId   map[string][]*tokenEntry     // protected by `mutex`
+	chains                map[vaa.ChainID]*chainEntry  // protected by `mutex`
+	msgsSeen              map[string]bool              // protected by `mutex` // Key is hash, payload is consts transferComplete and transferEnqueued.
```
Comment (SEJeff) [inline node/pkg/governor/governor.go:126]:
```suggestion
	msgsSeen              map[string]bool              // protected by `mutex`. Key is hash, payload is consts transferComplete and transferEnqueued.
```

Putting additional comment `//` in a comment is just very strange to me.

**Review by SEJeff — APPROVED** [review]

**Review by claudijd — APPROVED** [review]

**Pull request merged**